tests/PySys/mqtt_port/mqtt_port_change_connection_works/run.py

```diff
@@ -0,0 +1,132 @@
+import sys
+
+sys.path.append("environments")
+from environment_roundtrip_c8y import Environment_roundtrip_c8y
+
+from pysys.basetest import BaseTest
+
+import time
+
+"""
+Validate changing the mqtt port using the tedge command
+
+Given a configured system, that is configured with certificate created and registered in a cloud
+When the thin edge device is disconnected from cloud, sudo tedge disconnect c8y/az
+When mqtt.port is set using tedge with sudo
+When the thin edge device is connected to cloud, sudo tedge connect c8y/az
+Now validate the services that use the mqtt port
+   Validate tedge config set mqtt.port
+   Validate tedge mqtt pub/sub
+   Validate tedge connect c8y/az --test
+   Validate tedge_mapper status
+
+"""
+
+class MqttPortChangeConnectionWorks(Environment_roundtrip_c8y):
+    def setup(self):
+        self.tedge = "/usr/bin/tedge"
+        self.sudo = "/usr/bin/sudo"
+
+        # disconnect from c8y cloud
+        disconnect_c8y = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "disconnect", "c8y"],
+            stdouterr="disconnect_c8y",
+        )
+        self.addCleanupFunction(self.mqtt_cleanup)
+
+    def execute(self):
+        # set a new mqtt port for local communication
+        mqtt_port = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "config", "set", "mqtt.port", "8889"],
+            stdouterr="mqtt_port",
+        )
+
+        # wait for a while
+        time.sleep(0.1)
+
+        # connect to c8y cloud
+        connect_c8y = self.startProcess(
```

Comment:
Here we probably should run the connection test "tedge connect ... --test" to doulbecheck if we are really connected. That cool functionality should be also in the other tests...

Reply:
Thought of it, but felt its redundant, because already when the connection is established, it will give me the same log. I can use it if there is a strong reason for this.

Reply:
We kind of forgot to add the feature to the other test. We basically should check as tedge connect always returns 0. Nevertheless, I'll add a story to add the --test feature to other tests